pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/phimanifest/PhiManifestService.java

```diff
@@ -0,0 +1,131 @@
+package org.broadinstitute.dsm.service.phimanifest;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dao.mercury.MercuryOrderDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.dto.mercury.MercuryOrderDto;
+import org.broadinstitute.dsm.exception.DSMBadRequestException;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.model.elastic.Activities;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearchParticipantDto;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import spark.QueryParamsMap;
+
+@Slf4j
+public class PhiManifestService {
+
+    public Object generateReport(QueryParamsMap queryParams, String realm) {
+        String sequencingOrderNumber = queryParams.get(RoutePath.SEQ_ORDER_NUMBER).value();
+        String ddpParticipantId = queryParams.get(RoutePath.DDP_PARTICIPANT_ID).value();
+        DDPInstanceDto ddpInstanceDto = DDPInstanceDao.of().getDDPInstanceByInstanceName(realm).orElseThrow();
+        String maybeErrorMessage = null;
+        if (StringUtils.isBlank(ddpParticipantId)) {
+            maybeErrorMessage = "No DdpParticipantId was provided";
+        }
+        if (StringUtils.isBlank(sequencingOrderNumber)) {
+            maybeErrorMessage = "No sequencingOrderNumber was provided";
+        }
+        if (!isSequencingOrderValid(sequencingOrderNumber, ddpParticipantId, ddpInstanceDto)) {
+            maybeErrorMessage = String.format("Sequencing order number %s is not a valid order for participant %s ", sequencingOrderNumber,
+                    ddpParticipantId);
+        }
+        if (!isParticipantConsented(ddpParticipantId, ddpInstanceDto)) {
+            maybeErrorMessage = String.format("Participant %s has not consented to receive shared learning ", ddpParticipantId);
+        }
+        if (StringUtils.isNotBlank(maybeErrorMessage)) {
+            log.warn(maybeErrorMessage);
```

Comment:
Probably don't need a warn here--the exception thrown below will be handled/logged somewhere up the stack.